src/PluginClient.php

```diff
@@ -79,7 +78,7 @@ public function sendRequest(RequestInterface $request)
         $pluginChain = $this->createPluginChain($this->plugins, function (RequestInterface $request) {
             try {
                 return new FulfilledPromise($this->client->sendRequest($request));
-            } catch (HttplugException $exception) {
+            } catch (\Exception $exception) {
```

Comment:
Maybe there is a fourth option: add an Exception interface to Promise and accept only those in promises, every other exception needs to be thrown by definition. Then in plugins we still need to check whether the received exception is an Http one (Http ones should be Promise ones too). The difference is that this way in the promise handling we can force a type of exception without forcing `\Exception` (could be a solution for throwables too). It is a BC break from Promise point of view (exception typehints changed), however Promise is kind of an internal thing from httplug point of view, so I have no problem with tagging v2.